llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,231 @@
+                    document_id=document_id,
+                    metadata=source.metadata,
+                    client=client,
+                )
+
+            genaix.batch_create_chunk(
+                corpus_id=self.corpus_id,
+                document_id=document_id,
+                texts=[node.get_content() for node in nodeGroup.nodes],
+                metadatas=[node.metadata for node in nodeGroup.nodes],
+                client=client,
+            )
+
+        return [node.node_id for node in nodes]
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.delete(ref_doc_id={ref_doc_id})")
+
+        client = cast(genai.RetrieverServiceClient, self.client)
+        genaix.delete_document(
+            corpus_id=self.corpus_id, document_id=ref_doc_id, client=client
+        )
+
+    def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.query(query={query})")
+
+        client = cast(genai.RetrieverServiceClient, self.client)
+
+        relevant_chunks: List[genai.RelevantChunk] = []
+        if query.doc_ids is None:
+            relevant_chunks = genaix.query_corpus(
+                corpus_id=self.corpus_id,
+                query=query.query_str or "what?",
+                filter=_convertFilter(query.filters),
+                k=query.similarity_top_k,
+                client=client,
+            )
+        else:
+            for doc_id in query.doc_ids:
+                relevant_chunks.extend(
+                    genaix.query_document(
+                        corpus_id=self.corpus_id,
+                        document_id=doc_id,
+                        query=query.query_str or "what?",
+                        filter=_convertFilter(query.filters),
+                        k=query.similarity_top_k,
+                        client=client,
+                    )
+                )
+
+        return VectorStoreQueryResult(
+            nodes=[
+                TextNode(
+                    text=chunk.chunk.data.string_value,
+                    id_=_extract_chunk_id(chunk.chunk.name),
+                )
+                for chunk in relevant_chunks
+            ],
+            ids=[_extract_chunk_id(chunk.chunk.name) for chunk in relevant_chunks],
+            similarities=[chunk.chunk_relevance_score for chunk in relevant_chunks],
+        )
+
+
+def _extract_chunk_id(entity_name: str) -> str:
```

Comment:
Is there a style guide for lower camel vs lower snake case? Maybe should be consistent either way.